### DISCUSSION TITLE
construct explicit tetra

### DISCUSSION THREAD

**Discussion by alessiofumagalli** [nschloe/pygmsh#539] [Q&A]
Dear all,

in a 3d geometry, is it possible to construct by hand some of tetrahedra and let gmsh build the others, obtaining a conforming mesh? I'm using the python api.
Thanks a lot for your kind help,

Alessio

**Comment by nschloe** [comment]
Not with Gmsh, I don't think.

**Comment by alessiofumagalli** [comment]
I was using the gmsh.model.mesh.addElementsByType and get some results, but I have troubles making the generated grid conforming with the given tetra

**Comment by alessiofumagalli** [comment]
another possibility is to prevent gmsh to add new point on a given surface, with this I might be able to fix my issue
